### DISCUSSION TITLE
Update UI from non-ui thread

### DISCUSSION THREAD

**Discussion by mobynet1** [unoplatform/uno#18075] [General]
No matter what I try, I cannot get a simple textblock to update.

I have tried to update it from the view model:

```text
	var dispatcherQueue = Microsoft.UI.Dispatching.DispatcherQueue.GetForCurrentThread ();

	dispatcherQueue.TryEnqueue (Microsoft.UI.Dispatching.DispatcherQueuePriority.High, () => {
		AgentCheckMessage.Visibility = Visibility.Visible;
	});
```

Nope.

I moved this code into a button click handler directly in the view:

```text
	var dispatcherQueue = Microsoft.UI.Dispatching.DispatcherQueue.GetForCurrentThread ();

	dispatcherQueue.TryEnqueue (Microsoft.UI.Dispatching.DispatcherQueuePriority.High, () => {
		AgentCheckMessage.Visibility = Visibility.Visible;
	});
```

I know the button click runs in a non-UI thread, but why doesn't the TryEnque work even with high priority?

I have looked at examples in the Uno.Sample library, tried these approaches, yet nothing works. I know WPF wasn't this hard!

What am I doing wrong?

**Comment by MartinZikmund** [comment]
@mobynet1 generally button click does run on the UI thread, unless you are using MVUX where the threading is different.

I think the reason you may be facing issues is that you cannot call `DispatcherQueue.GetForCurrentThread` on a non-UI thread - instead, you must call it on the UI thread, store the instance in a field for example, and then use that cached instance when you want to enqueue the operation.

**Comment by mobynet1** [comment]
@MartinZikmund Thank you for such a rapid response! When I look at the name of the method, `GetForCurrentThread ` it would seem obvious that if I am not on the UI thread, which of course I'm not at this point, I wouldn't get what I was looking for! Doh! I need a vacation!!! :-)
